### PR TITLE
feat(sdkman): Create sdkman manager

### PR DESCRIPTION
**Pull request by momosetkn**
## Changes

support SDKMAN!

## Context

https://github.com/renovatebot/renovate/discussions/18835

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by momosetkn** [comment]
sorry, I don't know how to check the operation, and not supported other than java and build-tool.

i submitted pull-request. but, I want will close this PR, because, Node.js is not my specialty, I think it would be better to have someone else who is knowledgeable about renovate + Node.js do it.

File: lib/modules/manager/sdkman/datasources.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  // For most people, this setup will be enough.
+  switch (depName) {
+    case 'java':
+      return JavaVersionDatasource.id;
+    case 'maven':
+      return MavenDatasource.id;
+    case 'gradle':
+      return GradleVersionDatasource.id;
+    case 'sbt':
+      return ivyVersioning.id;
```
Comment (viceice) [inline lib/modules/manager/sdkman/datasources.ts:17]: wrong, don't use versioning as datasource

File: lib/modules/manager/sdkman/datasources.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { GradleVersionDatasource } from '../../datasource/gradle-version';
```
Comment (viceice) [inline lib/modules/manager/sdkman/datasources.ts:1]: rename file to `util`

File: lib/modules/manager/sdkman/__fixtures__/unknown-dep.sdkmanrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Enable auto-env through the sdkman_auto_env config
+# Add key=value pairs of SDKs to use below
+unknown=1.2.3
```
Comment (viceice) [inline lib/modules/manager/sdkman/__fixtures__/unknown-dep.sdkmanrc:3]: inline to test and use `codeBlock`, search codebase for samples

File: lib/modules/manager/sdkman/extract.spec.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          {
+            currentValue: '3.9.8',
+            datasource: 'maven',
+            depName: 'maven',
+            replaceString: 'maven= 3.9.8',
+          },
+          {
+            currentValue: '1.10.1',
+            datasource: 'ivy',
+            depName: 'sbt',
```
Comment (viceice) [inline lib/modules/manager/sdkman/extract.spec.ts:37]:
```suggestion
            datasource: 'github-releases',
            depName: 'sbt',
            packageName: 'sbt/sbt',
```

File: lib/modules/manager/sdkman/extract.spec.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          {
+            currentValue: '8.10',
+            datasource: 'gradle-version',
+            depName: 'gradle',
+            replaceString: 'gradle =8.10',
+          },
+          {
+            currentValue: '3.9.8',
+            datasource: 'maven',
+            depName: 'maven',
```
Comment (viceice) [inline lib/modules/manager/sdkman/extract.spec.ts:31]:
```suggestion
            depName: 'maven',
            packageName: 'org.apache.maven:maven',
```

File: lib/modules/manager/sdkman/extract.spec.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  describe('extractPackageFile()', () => {
+    it('extracts dependencies in a generic case', () => {
+      // when
+      const packageFile = extractPackageFile(genericCaseFileContent);
+
+      // then
+      expect(packageFile).toEqual({
+        deps: [
+          {
+            currentValue: '21.0.3-tem',
```
Comment (viceice) [inline lib/modules/manager/sdkman/extract.spec.ts:17]: this value is not supported by the java-version datasource

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by momosetkn** [comment]
@viceice 
Thank you very much for teaching.
I close ( or Temporarily close). because, renovate-datasource etc...It will needs major revision and Since the work needs to be done as we research, it will take time and the renovations cannot be completed immediately.

**Pull request closed without merging**